indico/legacy/pdfinterface/base.py

```diff
@@ -134,6 +135,7 @@ def setTTFonts():
         addMapping('LinuxLibertine', 1, 1, 'LinuxLibertine-Bold-Italic')
         pdfmetrics.registerFont(TTFont('Kochi-Mincho', os.path.join(font_dir, 'kochi-mincho-subst.ttf')))
         pdfmetrics.registerFont(TTFont('Kochi-Gothic', os.path.join(font_dir, 'kochi-gothic-subst.ttf')))
+        signals.legacy.register_additional_fonts.send()
```

Comment:
I'd rather have the signal return the font specs (name and path?) and do the registration here...

Reply:
Okay, but then how to go ahead with `addMapping`?

Reply:
@ThiefMaster,

-  I see `pdfmetrics.registerFont` calls `registerFontFamily(font.fontName)` so there's no need to explicitly do `addMapping` `from reportlab.lib.fonts import addMapping` ?

- Also, to return a name and path, do I just return a dictionary {font_name: path} from the function connected to the signal?

```python
plugin.connect(signals.legacy.register_additional_fonts, setAdditionalTTFonts)

def setAdditionalTTFonts(self):
    font_dir = os.path.join('../un/', 'indico_fonts') 
    return {'Bolt': os.path.join(font_dir, 'Bolt-Regular.ttf'), 'Bolt-Bold': os.path.join(font_dir, 'Bolt-Bold.ttf')}
```

Reply:
tbh, this is legacy code, I don't think anyone in the team remembers the details why these things were done that way :)